Add tests for AccumuloConfiguration.isPropertySet implementations

DomGarguilo opened the pull request:
Closes #3486 

This PR adds a new tests class with a test case for each impl of AccumuloConfiguration to verify that its implementation of `isPropertySet()` is behaving as expected.

ctubbsii reviewed (changes requested):
Found one issue. Didn't have time to finish looking, but the overall approach seems fine. But, please make sure you call EasyMock.verify on the mock objects.

keith-turner commented on server/base/src/test/java/org/apache/accumulo/server/conf/AccumuloConfigurationIsPropertySetTest.java:
```diff
@@ -0,0 +1,327 @@
+
+    Set<Property> shouldBeSet =
+        Set.of(INSTANCE_ZK_HOST, INSTANCE_SECRET, MANAGER_BULK_TSERVER_REGEX);
+    Set<Property> shouldNotBeSet = Sets.difference(shouldBeSet, ALL_PROPERTIES);
```
This needs the same change as mentioned elsewhere

```suggestion
    Set<Property> shouldNotBeSet = Sets.difference(ALL_PROPERTIES, shouldBeSet);
```

keith-turner commented on server/base/src/test/java/org/apache/accumulo/server/conf/AccumuloConfigurationIsPropertySetTest.java:
```diff
@@ -0,0 +1,327 @@
+    NamespaceConfiguration namespaceConfig =
+        new NamespaceConfiguration(context, nsid, parentConfig);
+
+    Set<Property> shouldBeSet = Set.of(TABLE_BLOOM_SIZE);
```
Seems like this set should be the following based on what is set for the parent and table conf object.

```suggestion
    Set<Property> shouldBeSet = Set.of(TABLE_BLOOM_SIZE, TABLE_DURABILITY, TABLE_BLOOM_ENABLED);
```

keith-turner commented on server/base/src/test/java/org/apache/accumulo/server/conf/AccumuloConfigurationIsPropertySetTest.java:
```diff
@@ -0,0 +1,327 @@
+        new NamespaceConfiguration(context, nsid, parentConfig);
+
+    Set<Property> shouldBeSet = Set.of(TABLE_BLOOM_SIZE);
+    Set<Property> shouldNotBeSet = Set.of(TSERV_SCAN_MAX_OPENFILES);
```
Could this do the set difference with all props that other test have done?

DomGarguilo commented:
Some of the tests are failing when i correct the Sets.difference() issue
mentioned in a previous comment. Will work to fix that tomorrow.

On Tue, Sep 5, 2023 at 4:56 PM EdColeman ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/accumulo/pull/3676#pullrequestreview-1611918044>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALMD2ILJXIOXWLLDGPIV4R3XY6GTHANCNFSM6AAAAAA3BVGDWY>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

DomGarguilo commented:
Correcting the usage of Sets.difference() in the remaining two test cases (in commit eceaedf) has caused `systemConfiguration()` and `siteConfuguration()` to fail.

The `siteConfuguration()` test fails the check where `getProperties()` is called on the object and many properties are set that are not expected to be. 

The `systemConfiguration` test is failing in a similar way except the `getProperties()` returns the expected properties but the `isPropertySet()` impl is indicating there are properties that are set that are not expected to be.

I am not super familiar with these two and I'm assuming there are some default properties that are included with the set of properties that are manually set upon initializing the object however I don't know where those are coming from or which props are expected to be there.

DomGarguilo commented:
Fixed the two test cases that were failing in bd3ef97 and bd3ef97. Should be ready for review now.

ctubbsii reviewed (approved):
@DomGarguilo I added a commit to this branch to improve the test. I kept all the test cases the same (well, some of the property values being set changed, but the actual values don't matter; that was just for convenience). But, I reorganized quite a bit of it.

I think it's ready to merge, but I'll leave it to you to go over what I changed, and determine if you agree with the changes I made before you merge it.

DomGarguilo commented:
> @DomGarguilo I added a commit to this branch to improve the test. I kept all the test cases the same (well, some of the property values being set changed, but the actual values don't matter; that was just for convenience). But, I reorganized quite a bit of it.
> 
> I think it's ready to merge, but I'll leave it to you to go over what I changed, and determine if you agree with the changes I made before you merge it.

The changes you made look great! I'll go ahead and merge this now.
